AdminServer/appscale/admin/__init__.py

```diff
@@ -270,11 +273,25 @@ def initialize(self, acc, ua_client, zk_client, version_update_lock,
   def get(self):
     """ List projects.
     """
-    self.authenticate()
-    try:
-      projects = self.zk_client.get_children('/appscale/projects')
-    except NoNodeError:
-      projects = []
+    if 'AppScale-Secret' not in self.request.headers:
+      # Run all the authenticate functions in authenticate_access_token but
+      # don't check if user is authenticated for a specific project,
+      # get projects user can access.
+      token = self.request.headers['Authorization'].split()[1]
+      method_base64, metadata_base64, signature = token.split('.')
+      self.check_token_hash(method_base64, metadata_base64, signature)
+
+      metadata = json.loads(base64.urlsafe_b64decode(metadata_base64))
+      self.check_token_expiration(metadata)
+      self.check_token_scope(metadata)
+      user = metadata['user']
+      if self.ua_client.is_user_cloud_admin(user):
```

Comment:
This should handle UAException.

Reply:
If we can't check if the user is cloud admin, should we assume they are not or should we raise an error?

Reply:
Since it is an operation error and does not have to do with the request, it should be an error in the 500 range.